core/iwasm/compilation/aot_emit_variable.c

```diff
@@ -149,57 +150,156 @@ aot_compile_op_tee_local(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
     return aot_compile_op_set_or_tee_local(comp_ctx, func_ctx, local_idx, true);
 }
 
+/*TODO: should be optimized by moving globals to WASMModuleInstance */
+LLVMValueRef
+get_global_from_wasm_inst(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+                          uint32 global_idx)
+{
+    /* WASMModuleInstance->e */
+    uint32 e_offset_val = offsetof(WASMModuleInstance, e);
```

Comment:
The `WASMModuleInstanceExtra` structure is put at the end of `WASMModuleInstance`, we can use a fixed offset to get it and its fields, like `offset_u32 = get_module_inst_extra_offset(comp_ctx)` and then `offset = offset_u32 + offsetof(AOTModuleInstanceExtra, globals)` for AOT or `offset = offset_u32 + offsetof(WASMModuleInstanceExtra, globals)` for JIT. By this way, we can save one load operation, refer to `create_shared_heap_info`:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_llvm.c#L1528-L1545